command/agent/txn_endpoint.go

```diff
@@ -0,0 +1,152 @@
+package agent
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/consul/structs"
+)
+
+// fixupKVOps takes the raw decoded JSON and base64 decodes values in KV ops,
+// replacing them with byte arrays.
+func fixupKVOps(raw interface{}) error {
+	// decodeValue decodes the value member of the given operation.
+	decodeValue := func(rawKV interface{}) error {
```

Comment:
Is there a reason these functions are defined locally. There could be some performance gains to having them statically defined and not in a dynamic variable. And this is in somewhat of a hot path.